src/opm/input/eclipse/Schedule/WellTraj/RigEclipseWellLogExtractor.cpp

```diff
@@ -0,0 +1,325 @@
+    //                                                                    hexCorners[faceVertexIndices[1]],
+    //                                                                    hexCorners[faceVertexIndices[2]],
+    //                                                                    hexCorners[faceVertexIndices[3]] );
+
+    cvf::Vec3d faceCenterCenterVectorI = faceCenterPosI - faceCenterNegI;
+    cvf::Vec3d faceCenterCenterVectorJ = faceCenterPosJ - faceCenterNegJ;
+
+    localZdirection.cross( faceCenterCenterVectorI, faceCenterCenterVectorJ );
+    localZdirection.normalize();
+
+    cvf::Vec3d crossPoductJZ;
+    crossPoductJZ.cross( faceCenterCenterVectorJ, localZdirection );
+    localXdirection = faceCenterCenterVectorI + crossPoductJZ;
+    localXdirection.normalize();
+
+    cvf::Vec3d crossPoductIZ;
+    crossPoductIZ.cross( faceCenterCenterVectorI, localZdirection );
+    localYdirection = faceCenterCenterVectorJ - crossPoductIZ;
+    localYdirection.normalize();
+}
+
+void RigEclipseWellLogExtractor::buildCellSearchTree()
+{
+    if (m_cellSearchTree.isNull()) {
+
+        auto nx = m_grid.getNX();
+        auto ny = m_grid.getNY();
+        auto nz = m_grid.getNZ();
+
+        size_t cellCount = nx * ny * nz;
+        std::vector<size_t>           cellIndicesForBoundingBoxes;
+        std::vector<cvf::BoundingBox> cellBoundingBoxes;
+
+// #pragma omp parallel
+//         {
+            size_t threadCellCount = cellCount;
+
+            std::vector<size_t>           threadIndicesForBoundingBoxes;
+            std::vector<cvf::BoundingBox> threadBoundingBoxes;
+
+            threadIndicesForBoundingBoxes.reserve( threadCellCount );
+            threadBoundingBoxes.reserve( threadCellCount );
+
+            std::array<double, 3> cornerPointArray;
+            cvf::Vec3d cornerPoint;
+
+// #pragma omp for
+            for (int cIdx = 0; cIdx < (int)cellCount; ++cIdx) {
+                const auto[i,j,k] = m_grid.getIJK(cIdx);
+                cvf::BoundingBox cellBB;
+                for (std::size_t l = 0; l < 8; l++) {
+                     cornerPointArray = m_grid.getCornerPos(i,j,k,l);
+                     cornerPoint = cvf::Vec3d(cornerPointArray[0], cornerPointArray[1], cornerPointArray[2]);
+                     cellBB.add(cornerPoint);
+                }
+
+                if (cellBB.isValid()) {
+                    threadIndicesForBoundingBoxes.emplace_back(cIdx);
+                    threadBoundingBoxes.emplace_back(cellBB);
+                }
+            }
+
+            threadIndicesForBoundingBoxes.shrink_to_fit();
+            threadBoundingBoxes.shrink_to_fit();
+// #pragma omp critical
+//          {
+             cellIndicesForBoundingBoxes.insert( cellIndicesForBoundingBoxes.end(),
+                                                 threadIndicesForBoundingBoxes.begin(),
+                                                 threadIndicesForBoundingBoxes.end() );
```

Comment:
I don't understand this construction.
As threadIndicesForBoundingBoxes is declared outside the for loop, never cleared, and entries are always added with emplace_back, I am assuming that cellIndicesForBoundingBoxes contains a lot of duplicates. I am even assuming that in the worst case it will contain cellCount*cellCount entries. Is this what we want? Or am I missing something here?